presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java

```diff
@@ -553,6 +558,26 @@ public PinotQueryGeneratorContext withVariablesInAggregation(Set<VariableReferen
                 useSqlSyntax);
     }
 
+    public PinotQueryGeneratorContext withDistinctLimit(LinkedHashSet<VariableReferenceExpression> newGroupByColumns, long limit)
+    {
+        int intLimit = checkForValidLimit(limit);
```

Comment:
Is it worth asserting that we are in sql mode here ?

Reply:
done.